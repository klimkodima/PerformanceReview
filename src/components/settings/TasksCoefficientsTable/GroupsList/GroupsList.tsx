import { ChangeEvent, ReactElement, memo } from 'react';

import { FormControlLabel, Box } from '@mui/material';
import ArrowRightIcon from '@mui/icons-material/ArrowRight';

import { WebsiteGroupsAccordion } from 'src/components/common';
import { GroupsListItem } from '../GroupsListItem';
import { CustomCheckbox } from '../WebsiteCheckbox';

import { ListWebsitesDataType, WebsiteGroupDataType } from './types';

import './GroupList.scss';

type GroupListPropsType = {
  isWebsitesOpen: boolean;
  websiteGroupData: WebsiteGroupDataType;
  ListWebsitesData: ListWebsitesDataType[];
  onOpenWebsitesClick: () => void;
  onWebsiteGroupClick: (event: ChangeEvent<HTMLInputElement>) => void;
  onWebsiteClick: (
    event: ChangeEvent<HTMLInputElement>,
    currentWebsiteName: string
  ) => void;
};

const GroupsList = memo(
  ({
    websiteGroupData,
    ListWebsitesData,
    onWebsiteGroupClick,
    onWebsiteClick,
    isWebsitesOpen,
    onOpenWebsitesClick
  }: GroupListPropsType): ReactElement => (
    <div className='group-list'>
      <WebsiteGroupsAccordion
        expanded={isWebsitesOpen}
        expandIcon={
          <ArrowRightIcon
            className='arrow-icon'
            onClick={onOpenWebsitesClick}
          />
        }
        summary={
          <FormControlLabel
            label={
              <span className='group-list-label' data-testid='group-list-label'>
                {websiteGroupData.websiteGroup}
              </span>
            }
            control={
              <CustomCheckbox
                checked={websiteGroupData.checked}
                onChange={onWebsiteGroupClick}
              />
            }
          />
        }
      >
        <Box className='group-list__sites'>
          {ListWebsitesData.map(({ websiteName }, index) => (
            <GroupsListItem
              key={websiteName}
              hasIndividualCoefficient={
                ListWebsitesData[index].hasIndividualCoefficient
              }
              label={websiteName}
              checked={ListWebsitesData[index].checked}
              onChange={onWebsiteClick}
            />
          ))}
        </Box>
      </WebsiteGroupsAccordion>
    </div>
  )
);

export default GroupsList;

GroupsList.displayName = 'GroupsList';
