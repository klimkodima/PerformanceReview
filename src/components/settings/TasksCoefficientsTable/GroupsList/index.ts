export { default as GroupsList } from './GroupsListContainer';
