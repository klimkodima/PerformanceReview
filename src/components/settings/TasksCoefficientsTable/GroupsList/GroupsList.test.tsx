import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { I18nextProvider } from 'react-i18next';
import i18n from 'src/i18n';

import store from 'src/store';
import { setCurrentEditData } from 'src/store/TasksCoeff';
import GroupsList from './GroupsList';
import GroupsListContainer from './GroupsListContainer';

beforeAll(() => {
  store.dispatch(
    setCurrentEditData({
      taskType: 'name',
      auditType: 'initial',
      groupCoefficient: 1,
      websiteGroup: 'OTA',
      websiteName: undefined,
      websiteCoefficient: undefined
    })
  );
});

const mockWebsite = [
  {
    websiteName: 'agoda.com',
    coefficients: {
      initial: { email: 0.6, fax: 0.25, name: 3 },
      reaudit: { address: 10, fax: 10, name: 4 }
    }
  },
  {
    websiteName: 'GPTS',
    coefficients: {
      initial: { name: 5, address: 1, web: 1, fax: 0.3, email: 0.25 },
      reaudit: { name: 6, address: 6, web: 6, fax: 6, email: 6 }
    }
  },
  {
    websiteName: '1siteWithGroupCoefficient.com'
  },
  {
    websiteName: '2siteWithGroupCoefficient.com'
  }
];

describe('GroupsList', () => {
  test('renders correctly', () => {
    const mockListWebsitesData = [
      {
        websiteName: 'some',
        checked: false,
        hasIndividualCoefficient: true
      },
      {
        websiteName: 'some2',
        checked: true,
        hasIndividualCoefficient: false
      }
    ];

    const mockWebsiteGroupData = {
      checked: false,
      websiteGroup: 'OTA'
    };

    const tree = renderer
      .create(
        <GroupsList
          onOpenWebsitesClick={() => true}
          onWebsiteClick={() => true}
          onWebsiteGroupClick={() => true}
          ListWebsitesData={mockListWebsitesData}
          isWebsitesOpen={true}
          websiteGroupData={mockWebsiteGroupData}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('container renders correctly', () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Provider store={store}>
            <GroupsListContainer website={mockWebsite} websiteGroup='OTA' />
          </Provider>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('container group checkbox click works', () => {
    render(
      <I18nextProvider i18n={i18n}>
        <Provider store={store}>
          <GroupsListContainer website={mockWebsite} websiteGroup='OTA' />
        </Provider>
      </I18nextProvider>
    );
    expect(screen.getAllByTestId('custom-checkbox')[0]).toHaveClass(
      'Mui-checked'
    );
    fireEvent.click(screen.getByText(/OTA/));
    expect(screen.getAllByTestId('custom-checkbox')[0]).not.toHaveClass(
      'Mui-checked'
    );
    fireEvent.click(screen.getByText(/OTA/));
    expect(screen.getAllByTestId('custom-checkbox')[0]).toHaveClass(
      'Mui-checked'
    );
  });

  test('container website checkbox click works', () => {
    render(
      <Provider store={store}>
        <GroupsListContainer website={mockWebsite} websiteGroup='OTA' />
      </Provider>
    );
    expect(screen.getAllByTestId('custom-checkbox')[1]).not.toHaveClass(
      'Mui-checked'
    );
    fireEvent.click(screen.getByText(/agoda.com/));
    expect(screen.getAllByTestId('custom-checkbox')[1]).toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox').length).toBe(5);
  });

  test('group checkbox select only checkbox without individual coefficient', () => {
    render(
      <Provider store={store}>
        <GroupsListContainer website={mockWebsite} websiteGroup='OTA' />
      </Provider>
    );
    expect(screen.getAllByTestId('custom-checkbox')[0]).toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[1]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[2]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[3]).toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[4]).toHaveClass(
      'Mui-checked'
    );
    fireEvent.click(screen.getByText(/agoda.com/));
    fireEvent.click(screen.getByText(/OTA/));
    expect(screen.getAllByTestId('custom-checkbox')[0]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[1]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[2]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[3]).not.toHaveClass(
      'Mui-checked'
    );
    expect(screen.getAllByTestId('custom-checkbox')[4]).not.toHaveClass(
      'Mui-checked'
    );
  });
});
