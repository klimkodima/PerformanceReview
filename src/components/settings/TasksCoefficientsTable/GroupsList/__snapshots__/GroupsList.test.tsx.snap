// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsList container renders correctly 1`] = `
<div
  className="group-list"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded custom-accordion css-1wc0u5o-MuiPaper-root-MuiAccordion-root"
    data-testid="custom-accordion"
  >
    <div
      className="custom-accordion__form"
    >
      <div
        aria-expanded={true}
        className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded custom-accordion-summary css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content Mui-expanded css-1betqn-MuiAccordionSummary-content"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              data-testid="custom-checkbox"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium custom-checkbox__check-icon css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              <span
                className="group-list-label"
                data-testid="group-list-label"
              >
                OTA
              </span>
            </span>
          </label>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowRightIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="m10 17 5-5-5-5v10z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            className="MuiAccordion-region"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root custom-accordion__details css-15v22id-MuiAccordionDetails-root"
            >
              <div
                className="group-list__sites MuiBox-root css-0"
              >
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      agoda.com
                      <svg
                        aria-hidden={true}
                        aria-label="Website with Individual coefficients"
                        aria-labelledby={null}
                        className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
                        data-mui-internal-clone-element={true}
                        data-testid="PriorityHighIcon"
                        focusable="false"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        viewBox="0 0 24 24"
                      >
                        <circle
                          cx="12"
                          cy="19"
                          r="2"
                        />
                        <path
                          d="M10 3h4v12h-4z"
                        />
                      </svg>
                    </span>
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      GPTS
                      <svg
                        aria-hidden={true}
                        aria-label="Website with Individual coefficients"
                        aria-labelledby={null}
                        className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
                        data-mui-internal-clone-element={true}
                        data-testid="PriorityHighIcon"
                        focusable="false"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        viewBox="0 0 24 24"
                      >
                        <circle
                          cx="12"
                          cy="19"
                          r="2"
                        />
                        <path
                          d="M10 3h4v12h-4z"
                        />
                      </svg>
                    </span>
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={true}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium custom-checkbox__check-icon css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      1siteWithGroupCoefficient.com
                    </span>
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={true}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium custom-checkbox__check-icon css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      2siteWithGroupCoefficient.com
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`GroupsList renders correctly 1`] = `
<div
  className="group-list"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded custom-accordion css-1wc0u5o-MuiPaper-root-MuiAccordion-root"
    data-testid="custom-accordion"
  >
    <div
      className="custom-accordion__form"
    >
      <div
        aria-expanded={true}
        className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded custom-accordion-summary css-1pvvkxv-MuiButtonBase-root-MuiAccordionSummary-root"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiAccordionSummary-content Mui-expanded css-1betqn-MuiAccordionSummary-content"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              data-testid="custom-checkbox"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                checked={false}
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              <span
                className="group-list-label"
                data-testid="group-list-label"
              >
                OTA
              </span>
            </span>
          </label>
        </div>
        <div
          className="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium arrow-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowRightIcon"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="m10 17 5-5-5-5v10z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
        >
          <div
            className="MuiAccordion-region"
            role="region"
          >
            <div
              className="MuiAccordionDetails-root custom-accordion__details css-15v22id-MuiAccordionDetails-root"
            >
              <div
                className="group-list__sites MuiBox-root css-0"
              >
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={false}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      some
                      <svg
                        aria-hidden={true}
                        aria-label="Website with Individual coefficients"
                        aria-labelledby={null}
                        className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
                        data-mui-internal-clone-element={true}
                        data-testid="PriorityHighIcon"
                        focusable="false"
                        onBlur={[Function]}
                        onFocus={[Function]}
                        onMouseLeave={[Function]}
                        onMouseOver={[Function]}
                        onTouchEnd={[Function]}
                        onTouchStart={[Function]}
                        viewBox="0 0 24 24"
                      >
                        <circle
                          cx="12"
                          cy="19"
                          r="2"
                        />
                        <path
                          d="M10 3h4v12h-4z"
                        />
                      </svg>
                    </span>
                  </span>
                </label>
                <label
                  className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd group-list-item css-j204z7-MuiFormControlLabel-root"
                >
                  <span
                    className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked custom-checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    data-testid="custom-checkbox"
                    onBlur={[Function]}
                    onContextMenu={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={null}
                  >
                    <input
                      checked={true}
                      className="PrivateSwitchBase-input css-1m9pwf3"
                      data-indeterminate={false}
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium custom-checkbox__check-icon css-i4bv87-MuiSvgIcon-root"
                      data-testid="CheckIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
                      />
                    </svg>
                    <span
                      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </span>
                  <span
                    className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                  >
                    <span
                      className="group-list-item-label"
                      data-testid="group-list-item-label"
                    >
                      some2
                    </span>
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
